giotto/mapper/visualization.py

```diff
@@ -15,36 +15,56 @@
-        The nerve of the refined pullback cover. Nodes correspond to cluster
-        sets, and an edge exists between two nodes if they share at least one
-        point in common.
-
-    data : ndarray of shape (n_samples, n_features)
-        The point cloud data used to generate the nerve.
-
-    node_pos : igraph.Graph.layout or ndarray of shape (n_nodes, n_dims)
-        Encodes the layout of the graph according to a layout algorithm or
-        pre-defined array of coordinates in an n-dimensional space.
-
-    node_color : ndarray of shape (n_nodes,)
-        The numerical values to color each node of the graph by.
-
-    columns_to_color : dict, optional, default: ``None``
-        Key-value pairs (column_name, column_index) to specify which columns of
-        :attr:`data` to color the graph by. Generates a dropdown widget to
-        select the columns to color by.
+    pipeline : :class:`MapperPipeline` object
+        Mapper pipeline to act on to data.
+
+    data : array-like of shape (n_samples, n_features)
+        Data used to generate the Mapper graph. Can be a pandas dataframe.
+
+    layout : None, str or callable, optional, default: ``'kamada-kawai'``
+        Layout algorithm for the graph. Can be any accepted value for the
+        ``layout`` parameter in the :meth:`layout` method of
+        :class:`igraph.Graph`. [1]_
+
+    layout_dim : int, default: ``2``
+        The number of dimensions for the layout. Can be 2 or 3.
+
+    color_variable : column index or name, or list of such, \
+        or ndarray/pandas dataframe of shape (n_samples, n_target_features), \
+        or (fit-)transformer object, or None, optional, default: ``None``
+        Specifies which quantity is to be used for node colouring. When it is
+        a numpy ndarray or pandas dataframe, it must have the same length as
+        `data` and is interpreted as a quantity of interest according to
+        which each node of the Mapper graph is to be coloured (see
+        :attr:`node_colors`). ``None`` is equivalent to passing `data`. If an
+        object implementing :meth:`transform` or :meth:`fit_transform`,
+        e.g. a scikit-learn estimator or pipeline, it is applied to `data`
+        to generate the quantity of interest. Otherwise, it must be a column
+        or subset of columns to be selected from `data`.
+
+    node_color_statistic : callable, or ndarray of shape (n_nodes,) or \
+        (n_nodes, 1), optional, default: ``numpy.mean``
+        Specifies how to determine the colors of each node. If a
+        numpy array, it must have the same length as the number of nodes in
+        the Mapper graph, and its values are used directly for node
+        coloring, ignoring `color_variable`. Otherwise, it must be a
+        callable object and is used to obtain a summary statistic within
+        each Mapper node of the quantity specified by :attr:`node_colors`.
+
+    color_by_columns_dropdown : bool, optional, default: ``True``
```

Comment:
Should the default be False, at the very least to avoid unnecessary computation in the current implementation (note: change both here and in the code)

Reply:
Agreed.

Reply:
Has this been done?

Reply:
Yes